### ISSUE TITLE
add parameter for guard object for router attribute macros

### ISSUE THREAD

**Issue by anatawa12** [actix/actix-web#2229]
Currently, `guard` parameter accepts guard function, not a guard object. I want to use guard object like `Header("", "")` in router attribute. So, I request adding a parameter for `guard_obj` for router attribute macros.

**Comment by Silentdoer** [comment]
agree, this is useful, it can be obj_guard(like fn_guard) solved, obj_guard's value like guard::Get(), guard::Host(...),...
and maybe guard in macro attribue should rename to guard_fn?,such as #[route("/multi", guard_fn="xxx"]
